### ISSUE TITLE
Update infra to enable Heroku review apps on PRs

### ISSUE THREAD

**Issue by carols10cents** [rust-lang/crates.io#1710]
https://devcenter.heroku.com/articles/github-integration-review-apps

We can't enable review apps automatically:

> For security and billing reasons, Heroku does not automatically create review apps for pull requests to public repos that are sent from forks. You can still create review apps for these pull requests manually.

I haven't found yet in my quick skim of the docs what permissions someone needs to manually create a review app. If you need heroku permissions, maybe this is something we could have a bot do if this proves to be useful.

Review apps use app.json, which we have, but I suspect our current one is out of date. Also, it's set up to make it easy to deploy a mirror, and we'd probably want to have a small database to enable testing of publishing instead. Not sure yet how to have one app.json for the heroku button in the README and one for review apps... maybe we can get rid of the heroku button? Not sure anyone actually uses it.

**Comment by syphar** [comment]
Is this still something we want? 
Just stumbled on this, and having quite some Heroku experience I could check what's possible. 

I guess you need a real Heroku Account in the team to be able to directly create review apps, 
though It could be possible doing that via a GH action (based on a label? specific comment? ) or as a real bot.

**Comment by syphar** [comment]
it also should be possible to have one app.json work for both the review apps and the button, 

the most tricky thing is often to create the test-data, not sure how you do that currently.

**Comment by Turbo87** [comment]
now that the app uses DataDog for logging and CrunchyBridge for the database I'm not sure how viable such review apps will be. aside from that the cost of such review apps will probably also be an issue. and then there is finally the issue of needing to have a filled database to be able to review a significant portion of PR, so we would potentially have to clone the production database (?) which then has security implications.

tl;dr it sounds good in theory, but seems hard to pull off in practice and is likely not worth the effort
